src/app/test/api/http/unit/authentication/login_handlers/ldap_test.py

```diff
@@ -0,0 +1,509 @@
+        mock_connection.return_value = no_user
+        handler = LdapLoginHandler()
+
+        request = HTTPServerRequest(
+            body=b'{"username":"baduser","password":"not_user1"}'
+        )
+        authenticated_user = handler.get_user(request)
+
+        assert authenticated_user is None
+
+    @patch(
+        "beer_garden.api.http.authentication.login_handlers.ldap.LdapLoginHandler.get_connection"
+    )
+    def test_no_added_group(self, mock_connection, ldap_connection):
+        mock_connection.return_value = ldap_connection
+        handler = LdapLoginHandler()
+
+        request = HTTPServerRequest(body=b'{"username":"jsmith1","password":"jsmith1"}')
+        authenticated_user = handler.get_user(request)
+
+        assert authenticated_user is not None
+        assert authenticated_user.username == "jsmith1"
+        assert len(authenticated_user.roles) == 0
+
+    @patch(
+        "beer_garden.api.http.authentication.login_handlers.ldap.LdapLoginHandler.get_connection"
+    )
+    def test_default_matches_group(self, mock_connection, ldap_connection, role1):
+        mock_connection.return_value = ldap_connection
+        handler = LdapLoginHandler()
+
+        request = HTTPServerRequest(body=b'{"username":"jsmith1","password":"jsmith1"}')
+        authenticated_user = handler.get_user(request)
+
+        assert authenticated_user is not None
+        assert authenticated_user.username == "jsmith1"
+        assert len(authenticated_user.roles) == 1
+
+    @patch(
+        "beer_garden.api.http.authentication.login_handlers.ldap.LdapLoginHandler.get_connection"
+    )
+    def test_added_group(self, mock_connection, ldap_connection, role1, role2):
+        mock_connection.return_value = ldap_connection
+        handler = LdapLoginHandler()
+
+        request = HTTPServerRequest(body=b'{"username":"jsmith1","password":"jsmith1"}')
+        authenticated_user = handler.get_user(request)
+
+        assert authenticated_user is not None
+        assert authenticated_user.username == "jsmith1"
+        assert len(authenticated_user.roles) == 2
+
+    @patch(
+        "beer_garden.api.http.authentication.login_handlers.ldap.LdapLoginHandler.get_connection"
+    )
+    def test_user_groups_replaced(
+        self, mock_connection, ldap_connection2, role0, role2, role3, role4
+    ):
+        user = create_user(User(username="sbrown20", roles=[role0]))
+        assert user is not None
+        assert user.username == "sbrown20"
+        assert len(user.local_roles) == 1
+
+        mock_connection.return_value = ldap_connection2
+        handler = LdapLoginHandler()
+
+        request = HTTPServerRequest(
+            body=b'{"username":"sbrown20","password":"sbrown20"}'
+        )
+        authenticated_user = handler.get_user(request)
+
+        assert authenticated_user is not None
+        assert authenticated_user.username == "sbrown20"
+        assert len(authenticated_user.local_roles) == 2
```

Comment:
Along with asserting the length, we want to confirm the expected roles were added. Could you add this to the section after the len assertion:
```python
        for local_role in authenticated_user.local_roles:
            if local_role.name == role2.name or local_role.name == role3.name:
                # Confirm that new roles were added
                assert True
            elif  local_role.name == role1.name:
                # Confirm that old role was removed
                assert False 
            else:
                # Unknown role added
                assert False
```